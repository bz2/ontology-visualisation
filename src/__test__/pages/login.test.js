import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Login from '../../pages/login'

const setup = ({
  activeLoaders
}) => {
  const props = {
    addNumber: jest.fn(),
    addToObject: jest.fn(),
    activeLoaders
  }

  const component = shallow(<Login {...props} />)

  return {
    component
  }
}

describe('Login page', () => {
  it('should match snapshot when activeLoaders', () => {
    const { component } = setup({
      activeLoaders: 1
    })
    expect(toJson(component)).toMatchSnapshot()
  })

  it('should match snapshot when no activeLoaders', () => {
    const { component } = setup({
      activeLoaders: 0
    })
    expect(toJson(component)).toMatchSnapshot()
  })
})
