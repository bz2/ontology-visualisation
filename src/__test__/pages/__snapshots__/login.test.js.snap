// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login page should match snapshot when activeLoaders 1`] = `
<Connect
  actions={
    Object {
      "addNumber": [Function],
      "addSubValueToObject": [Function],
      "addToArray": [Function],
      "addToObject": [Function],
      "removeFromArray": [Function],
      "removeFromObject": [Function],
      "setStoreState": [Function],
    }
  }
  activeLoaders={1}
  addNumber={[MockFunction]}
  addToObject={[MockFunction]}
  mapToProps={[Function]}
>
  <Component />
</Connect>
`;

exports[`Login page should match snapshot when no activeLoaders 1`] = `
<Connect
  actions={
    Object {
      "addNumber": [Function],
      "addSubValueToObject": [Function],
      "addToArray": [Function],
      "addToObject": [Function],
      "removeFromArray": [Function],
      "removeFromObject": [Function],
      "setStoreState": [Function],
    }
  }
  activeLoaders={0}
  addNumber={[MockFunction]}
  addToObject={[MockFunction]}
  mapToProps={[Function]}
>
  <Component />
</Connect>
`;
